File: backend-api/app/Models/Professeur.php
<?php

namespace App\Models;

use Illuminate\Foundation\Auth\User as Authenticatable;

// On étend Authenticatable au lieu de Model
// car ce modèle représente un utilisateur qui peut se connecter
class Professeur extends Authenticatable
{
    // Colonnes autorisées à être remplies via l'API
    protected $fillable = [
        'nom',
        'prenom',
        'email',
        'password',
    ];

    // Colonnes sensibles cachées
    // Ces champs ne seront JAMAIS retournés dans les réponses JSON
    protected $hidden = [
        'password',
    ];

    // Un professeur peut enseigner plusieurs matières (max 2)
    public function matieres()
    {
        return $this->belongsToMany(Matiere::class, 'professeur_matiere');
    }
}

File: mobile/lib/models/professeur.dart
import 'matiere.dart';

// Modèle représentant un professeur
class Professeur {
  final int id;
  final String nom;
  final String prenom;
  final String email;
  final List<Matiere> matieres;

  Professeur({
    required this.id,
    required this.nom,
    required this.prenom,
    required this.email,
    this.matieres = const [],
  });

  factory Professeur.fromJson(Map<String, dynamic> json) {
    return Professeur(
      id: json['id'],
      nom: json['nom'],
      prenom: json['prenom'],
      email: json['email'],
      // "matieres" sans accent pour correspondre à la réponse de l'API
      matieres: json['matieres'] != null
          ? (json['matieres'] as List)
              .map((m) => Matiere.fromJson(m))
              .toList()
          : [],
    );
  }
}